src/Common/src/System/Runtime/InteropServices/FixedString.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Contains definitions for various fixed size strings for creating blittable
+    /// structs. Provides easy string property access. Set strings are always null
+    /// terminated and will truncate if too large.
+    /// 
+    /// Usage: Instead of "fixed char _buffer[12]" use "FixedBuffer.Size12 _buffer"
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+    internal unsafe struct FixedString
```

Comment:
@jaredpar This is the struct we discussed- is this along the lines of what you had done? Do you have any suggestions for improvements?